### PR TITLE
Add documentation for Python and Rust

### PR DESCRIPTION
**Pull request by glokta1**

### CODE REVIEW

**Comment by glokta1** [comment]
@rohitpaulk I noticed the Go documentation has a leading `$` in every shell command. This could be annoying for people who're trying to copy the commands as the `$` gets copied too. Should I remove it?

File: docs/challenges/language-support/python.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+---
+
+<Note>
+  **New to Python?** We recommend checking out [Automate the Boring Stuff with
+  Python](https://automatetheboringstuff.com/) to get up to speed.
+</Note>
+
+# Local development
+
+To run the code for your challenge locally, you'll need `Python` installed. Our test runners use `Python` version `3.12` (as of April 2024).
```
Comment (rohitpaulk) [inline docs/challenges/language-support/python.mdx:13]:
```suggestion
To run the code for your challenge locally, you'll need `python3` installed. Our test runners use version `3.12` (as of April 2024).
```

(The value here corresponds to what CLI command they need to have - you'll see that the Java article mentions `mvn` too)

File: docs/challenges/language-support/python.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+The script for your challenge (like `./spawn_redis_server.sh`) will run the code using this version of the Python interpreter. It looks something like this:
+
+```sh
+#!/bin/sh
+
+exec python3 -m app.main "$@"
```
Comment (rohitpaulk) [inline docs/challenges/language-support/python.mdx:20]: Hmm, yeah it's a bit weird that this script differs - for a challenge that supports pipenv it is different, and for challenges that don't it is different. This is fine for now, we'll standardize these and then come back and update this.

File: docs/challenges/language-support/python.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+For example, if you added a file at `app/foo.py`, you could use it like so:
+
+<CodeGroup>
+```Python app/foo.py
+def foo():
+    print("Hello world!")
+````
+
+```Python app/main.py
+
```
Comment (rohitpaulk) [inline docs/challenges/language-support/python.mdx:41]: Extra space at top of file

File: docs/challenges/language-support/rust.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+<Note>
+  **New to Rust?** We recommend checking out [Rust by
+  Example](https://doc.rust-lang.org/rust-by-example/) to get up to speed.
+</Note>
+
+# Local development
+
+To run the code for your challenge locally, you'll need `Rust` installed. Our test runners use version `1.76` (as of April 2024).
```
Comment (rohitpaulk) [inline docs/challenges/language-support/rust.mdx:13]:
```suggestion
To run the code for your challenge locally, you'll need `cargo` installed. Our test runners use version `1.76` (as of April 2024).
```

File: docs/challenges/language-support/rust.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+mod foo;
+
+fn main() {
+  foo:foo(); // Calls the foo function from foo.rs
+}
+```
+
+</CodeGroup>
+
+<Note>
```
Comment (rohitpaulk) [inline docs/challenges/language-support/rust.mdx:57]: Good note!

File: docs/challenges/language-support/rust.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  Remember to make `foo()` accessible to `main.rs` by adding the `pub`
+  identifier
+</Note>
+
+# Adding dependencies
+
+Since version 1.62, Rust allows adding dependencies using `cargo add`.
```
Comment (rohitpaulk) [inline docs/challenges/language-support/rust.mdx:64]:
```suggestion
You can add dependencies to your program using `cargo add`.
```

(I don't think the 1.62 thing is relevant here, we're always dealing with newer versions)

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**